### PR TITLE
feat: add workspace support

### PR DESCRIPTION
**Pull request by fritzy**
* optional configuration to include root and workspaces
* github action for changes in workspace

### CODE REVIEW

File: bin/npm-template-check.js [hunk 1]
```diff
@@ -12,10 +13,15 @@ const main = async () => {
...
-  ]
+  const config = await getConfig(root)
+
+  const problemSets = []
+  for (const path of config.paths) {
+    problemSets.push(await checkPackage(path))
+    problemSets.push(await checkGitIgnore(path))
+  }
+
+  const problems = problemSets.flat()
```
Comment (wraithgar) [inline bin/npm-template-check.js:24]: are checkPackage or checkGitIgnore returning arrays?

File: lib/content/ci-workspace.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      - latest
+  workflow_dispatch:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the npm/cli repo
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14.x
```
Comment (wraithgar) [inline lib/content/ci-workspace.yml:23]: we lint on 16 in all the other repos

File: lib/postinstall/copy-content.js [hunk 3]
```diff
@@ -1,44 +1,43 @@
 const { dirname, join, resolve } = require('path')
 const fs = require('@npmcli/fs')
+const { readFile, writeFile } = require('fs/promises')
+const PackageJson = require('@npmcli/package-json')
 
 const contentDir = resolve(__dirname, '..', 'content')
 
 // keys are destination paths in the target project
 // values are paths to contents relative to '../content/'
 const content = {
```
Comment (wraithgar) [inline lib/postinstall/copy-content.js:10]: if the other list is `rootContent` what is this one?  Probably need a more descriptive name here than just `content`

File: test/postinstall/copy-content.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ const copyContent = require('../../lib/postinstall/copy-content.js')
 t.test('copies content', async (t) => {
   const root = t.testdir()
 
-  await copyContent(root)
+  await copyContent(root, root)
```
Comment (wraithgar) [inline test/postinstall/copy-content.js:10]: ???

File: test/postinstall/copy-content.js [hunk 5]
```diff
@@ -29,11 +29,15 @@ t.test('removes files', async (t) => {
   ]
   const root = t.testdir(content)
 
-  await copyContent(root)
+  await copyContent(root, root)
```
Comment (wraithgar) [inline test/postinstall/copy-content.js:32]: ???

File: bin/npm-template-check.js (hunk 1 above)
Comment (fritzy) [reply bin/npm-template-check.js:24]: Yes.

File: lib/content/ci-workspace.yml (hunk 2 above)
Comment (fritzy) [reply lib/content/ci-workspace.yml]: Fixed.

File: lib/postinstall/copy-content.js (hunk 3 above)
Comment (fritzy) [reply lib/postinstall/copy-content.js]: renamed `content`and `rootContent` to `moduleFiles` and `repoFiles`, respectively.

**Review by wraithgar — APPROVED** [review]

**Review by wraithgar — APPROVED** [review]

**Pull request merged**